drools-parser/src/main/antlr4/org/drools/parser/DRL6Expressions.g4

```diff
@@ -397,12 +404,14 @@ scope { BaseDescr lsd; }
           } else {
               $result = $left.result;
           }
+          // TODO access lsd directly instead of through dynamic context here
           $relationalExpression::lsd = $result;
       }
     }
-  ( ( operator | LEFT_PAREN )=> right=orRestriction
+  ( right=orRestriction
```

Comment:
This is incorrect and causes a failure in the dumper test.

Reply:
The failure was actually caused by something else. The operator rule couldn't match because it expected tokens such as STRING_LITERAL and IDENTIFIER (that could be recognized as an operator using a semantic predicate) while in fact the token stream contained unexpected tokens such as DRL_STRING_LITERAL and DRL_MATCHES that were introduced in the new DRL Lexer _before_ the DRL Expression parses migration started, which is why the new tokens haven't caused any issues up until now.